notify/grafana_alertmanager.go

```diff
@@ -45,21 +42,10 @@ const (
 )
 
 func init() {
-	silence.ValidateMatcher = func(m *pb.Matcher) error {
-		switch m.Type {
-		case pb.Matcher_EQUAL, pb.Matcher_NOT_EQUAL:
-			if !model.LabelValue(m.Pattern).IsValid() {
-				return fmt.Errorf("invalid label value %q", m.Pattern)
-			}
-		case pb.Matcher_REGEXP, pb.Matcher_NOT_REGEXP:
-			if _, err := regexp.Compile(m.Pattern); err != nil {
-				return fmt.Errorf("invalid regular expression %q: %s", m.Pattern, err)
-			}
-		default:
-			return fmt.Errorf("unknown matcher type %q", m.Type)
-		}
-		return nil
-	}
+	// This initializes the compat package in fallback mode. It parses first using the UTF-8 parser
+	// and then fallsback to the classic parser on error. UTF-8 is permitted in label names.
+	// This should be removed when the compat package is removed from Alertmanager.
+	compat.InitFromFlags(log.NewNopLogger(), compat.RegisteredMetrics, featurecontrol.NoopFlags{})
```

Comment:
The reason we need to use fallback mode to start with is that Grafana can manage configurations in Cortex and Mimir, and read configurations from Prometheus Alertmanager. Until most of those installations are using UTF-8 mode, Grafana will need to operate in fallback mode. I don't think we can take a %, but instead change it after some period of time (like 6 months).